02-microarray/gene-id-annotation_microarray_01_ensembl.Rmd

```diff
@@ -0,0 +1,368 @@
+
+Data downloaded from refine.bio include a metadata tab separated values (TSV) file and a data TSV file. 
+This chunk of code will read the both TSV files and add them as data frames to your environment.
+
+```{r}
+# Read in metadata TSV file
+metadata <- readr::read_tsv(file.path(
+  data_dir, # Replace with path to your metadata file
+  "metadata_GSE13490.tsv" # Replace with the name of your metadata file
+))
+
+# Read in data TSV file
+df <- readr::read_tsv(file.path(
+  data_dir, # Replace with path to your data file
+  "GSE13490.tsv" # Replace with the name of your data file
+)) %>%
+  # Tuck away the Gene ID column as rownames
+  tibble::column_to_rownames("Gene")
+```
+
+Let's ensure that the metadata and data are in the same sample order. 
+
+```{r}
+# Make the data in the order of the metadata
+df <- df %>%
+  dplyr::select(metadata$geo_accession)
+
+# Check if this is in the same order
+all.equal(colnames(df), metadata$geo_accession)
+
+# Bring back the "Gene" column in preparation for mapping
+df <- df %>%
+  tibble::rownames_to_column("Gene")
+```
+
+
+## Map Ensembl IDs to gene symbols
+
+This code chunk will return a message that indicates that there are many mappings between our keys and columns but this is okay and is what we expect in most cases since genes and transcripts do not have a 1:1 relationship.
+The `mapIds()` function has a `multiVals` argument which denotes what to do when there are multiple mapped values for a single gene identifier.
+The default behavior is to return just the first mapped value.
+It is good to keep in mind that various downstream analyses may benefit from varied strategies at this step. 
+Use `?mapIds` to see more options or strategies.
+
+In the next chunk, we will run the `mapIds()` function and supply the `multiVals` argument with the `"list"` option in order to get a large list with all the mapped values found for each gene identifier.
+
+```{r}
+# Map ensembl IDs to their associated gene symbols
+mapped_list <- mapIds(
+  org.Mm.eg.db, # Replace with annotation package for the organism relevant to your data
+  keys = df$Gene,
+  column = "SYMBOL", # Replace with the type of gene identifiers you would like to map to
+  keytype = "ENSEMBL", # Replace with the type of gene identifiers in your data
+  multiVals = "list"
+)
+```
+
+## Explore gene ID conversion
+
+Now, let's take a look at our mapped object to see how the mapping went.
+
+```{r}
+# Let's use the `head()` function to take a preview at our mapped list
+head(mapped_list)
+```
+
+It looks like we have gene symbols that were successfully mapped to the Ensembl IDs we provided.
+However, the data is now in a `list` object, making it a little more difficult to explore.
+We are going to turn our list object into a data frame object in the next chunk.
+
+```{r}
+# Let's make our object a bit more manageable for exploration by turning it into a data frame using the `reshape2::melt()` function
+mapped_df <- mapped_list %>%
+  reshape2::melt()
```

Comment:
I *think* you can use the `value` arugment to name what you'd like the column to be called, but you'd have to try that. I think we should either do that or use `rename()` so its clear what these columns are.